src/Corax/IndexWriter.cs

```diff
@@ -639,22 +619,17 @@ public unsafe long Update(string field, Span<byte> key, LazyStringValue id, Span
             oldEntryReader = default; // this is here to ensure that no one is going to try to *use* the old reader after we modify it
 
             // can't fit in old size, have to update the location
-            int reqSize = idLen + id.Size + data.Length;
-            if (rawSize < reqSize)
+            if (rawSize != data.Length)
             {
                 // remove the existing entry
                 Container.Delete(Transaction.LowLevelTransaction, _entriesContainerId, entryContainerId);
```

Comment:
In case this will be valid after your newest work:
Shouldn't be:
`rawData < data.Length`
In case when the old container is bigger than the new data we can fit it into the previous location.

Reply:
No, since the entries becomes a lot smaller, the cost of adding the size of the entry to the data is actually expensive.
To give some context, we can now hold an entry in < 16 bytes including the container overhead.